### ISSUE TITLE
8.2.0 breaks extract

### ISSUE THREAD

**Issue by vandres** [vendure-ecommerce/ngx-translate-extract#14]
For me, an update to 8.2.0 breaks everything.

This command now empties the files on the first execute without merging. On the second execute, it creates these file with the source strings.

`ngx-translate-extract -k --input ./src --output ./src/static/admin/assets/i18n/{en,de,sv,ja,es,fr,it,nl}.json --clean --sort --format json`

In another project, which works with PO files, I get an error `An error occurred: TypeError: Cannot read properties of undefined (reading 'reference')`

`ngx-translate-extract -k --input ./src --output ./src/static/pendla/assets/i18n/{en,de,fr,it,nl,es,sv,ja}.po --clean --sort --format pot`

Going back to 8.1.1 makes it work again.

**Comment by veneliniliev** [comment]
I have the same problem :(

**Comment by aevtimov** [comment]
The same it for me

**Comment by michaelbromley** [comment]
Hi all. Sorry about this break. I'm guessing it has to do with the changes in this PR, since it was the only change since 8.1.1

- https://github.com/vendure-ecommerce/ngx-translate-extract/pull/9

@SamanthaAdrichem do you have any idea what might be going on here?

**Comment by MMatsovIV** [comment]
Hey @SamanthaAdrichem, I think the problem comes with the update of the `TranslationType` interface. I know it's not much information but hope it will help you a bit. I will try to find some more time to spend on the issue as well.

**Comment by SamanthaAdrichem** [comment]
Apparently the JSON parser wasn't in the tests. But even if it was, due to lack of type hinting (which I could not completely fix either XD) I never saw the parse function wasn't parsing it into the new model 😓 sorry for that.

I think the attached pull request should fix the issues. Can someone confirm?

Just clone the pull request, run build and then run your script with node for example with the first guys comment after cloning (assuming you put the folder next to your project folder like so
```
.
..
ngx-translate-extract/
your-project/
```

run it
```
node ../ngx-translate-extract/dist/cli/cli.js -k --input ./src --output ./src/static/admin/assets/i18n/{en,de,sv,ja,es,fr,it,nl}.json --clean --sort --format json
```

Same with the PO one
```
node ../ngx-translate-extract/dist/cli/cli.js -k --input ./src --output ./src/static/pendla/assets/i18n/{en,de,fr,it,nl,es,sv,ja}.po --clean --sort --format pot
```

**Comment by vandres** [comment]
> Sorry that I couldn't look at it last Friday :(
> 
> Apparently the JSON parser wasn't in the tests. But even if it was, due to lack of type hinting (which I could not completely fix either XD) I never saw the parse function wasn't parsing it into the new model 😓 sorry for that.
> 
> I think the attached pull request should fix the issues. Can someone confirm?
> 
> Just clone the pull request,
> 
> ```
> git clone git@github.com:SamanthaAdrichem/ngx-translate-extract.git ./ngx-translate-extract
> cd ngx-translate-extract
> git checkout issue-14
> ```
> 
> Then run build and then run your script with node for example with the first guys comment after cloning (assuming you put the folder next to your project folder like so)
> 
> ```
> .
> ..
> ngx-translate-extract/
> your-project/
> ```
> 
> run it
> 
> ```
> node ../ngx-translate-extract/dist/cli/cli.js -k --input ./src --output ./src/static/admin/assets/i18n/{en,de,sv,ja,es,fr,it,nl}.json --clean --sort --format json
> ```
> 
> Same with the PO one
> 
> ```
> node ../ngx-translate-extract/dist/cli/cli.js -k --input ./src --output ./src/static/pendla/assets/i18n/{en,de,fr,it,nl,es,sv,ja}.po --clean --sort --format pot
> ```

In the source is no dist folder, I was able to build the project though. It seems to work now, thank you very much!

**Comment by SamanthaAdrichem** [comment]
It still needs to be merged and released, closing this might prevent that from happening.

**Comment by michaelbromley** [comment]
Just published v8.2.1 🥳
